# Оглавление
1. [Описательная статистика. Качественные и количественные характеристики популяции. Графическое представление данных](#discrete)
    * [Задание 1](#statistics_1)
    * [Задание 2](#statistics_2)
    * [Задание 3](#statistics_3)
        * [Задание 3.а](#statistics_3_1)
        * [Задание 3.б](#statistics_3_2)
        * [Задание 3.в](#statistics_3_3)
    * [Задание 4](#statistics_4)
        * [Задание 4.а](#statistics_4_1)
        * [Задание 4.б](#statistics_4_2)
        * [Задание 4.в](#statistics_4_3)
    * [Задание 5](#statistics_5)
        * [Задание 5.а](#statistics_5_1)
        * [Задание 5.б](#statistics_5_2)
        * [Задание 5.в](#statistics_5_3)
        * [Задание 5.г](#statistics_5_4)

# Описательная статистика. Качественные и количественные характеристики популяции. Графическое представление данных <a class='anchor' id='statistics'>

In [1]:
from math import factorial
import numpy as np

Количество **сочетаний** $k$ элементов, выбранных из $n$ элементов, которое вычисляется по формуле:

$$C_n^k = \frac{n!}{k!(n - k)!}.$$

In [2]:
def combinations(n, k):
    """ 
    Вычисляет количество сочетаний k элементов из n
    :param n: общее количество элементов
    :param k: количество выбранных элементов
    :rtype: int
    """
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

Чтобы определить вероятность события $B$ при условии, что событие $A$ уже произошло, будем использовать **формулу Байеса**:

$$P(B|A) = \frac{P(B) \cdot P(A|B)}{P(A)},$$

где вероятность события $A$ вычисляется по **формуле полной вероятности**:

$$P(A) = \sum_{i = 1}^N P(B_i) \cdot P(A|B_i),$$

события $B_i$, $i=1,.., N$ - полная группа событий, событие $A$ может наступить только при появлений событий $B_i$.

In [3]:
def bayes(P_A, P_A_B, P_B):
    """ 
    Вычисляет апостериорную вероятность события B при условии, что событие A уже произошло
    :param P_A: вероятность события A
    :param P_A_B: вероятность события A при условии B
    :param P_B: вероятность события B, определенная до испытания
    :rtype: float
    """
    return (P_B * P_A_B) / P_A

### Задание 1 <a class='anchor' id='statistics_1'>
    
Даны значения зарплат из выборки выпускников: 100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150. 

*Посчитать (желательно без использования статистических методов наподобие std, var, mean) среднее арифметическое, среднее квадратичное отклонение, смещенную и несмещенную оценки дисперсий для данной выборки.*

In [4]:
arr = np.array([100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150])

**Среднее арифметическое**

In [5]:
# Расчет через формулы
mean = np.sum(arr) / arr.size
mean

65.3

In [6]:
# Проверка через numpy
np.mean(arr)

65.3

**Смещенная оценка дисперсии**

In [7]:
# Расчет через формулы
var = np.sum(np.power(arr - mean, 2)) / arr.size
var

950.11

In [8]:
# Проверка
np.var(arr)

950.11

**Несмещенная оценка дисперсии**

In [9]:
# Расчет через формулы
var_ddof = np.sum(np.power(arr - mean, 2)) / (arr.size - 1)
var_ddof

1000.1157894736842

In [10]:
# Проверка
np.var(arr, ddof=1)

1000.1157894736842

**Среднее квадратичное отклонение (смещенное)**

In [11]:
# Расчет через формулы
std = np.sqrt(var)
std

30.823854398825596

In [12]:
np.std(arr)

30.823854398825596

**Среднее квадратичное отклонение (несмещенное)**

In [13]:
# Расчет через формулы
std_ddof = np.sqrt(var_ddof)
std_ddof

31.624607341019814

In [14]:
# Проверка
np.std(arr, ddof=1)

31.624607341019814

**Ответ (задание 1):**

Мат. ожидание = 65.3

Дисперсия (смещенная оценка) = 950.11

Дисперсия (несмещенная оценка) $\approx$ 1000.116

СКО (смещенная оценка) $\approx$ 30.824

СКО (несмещенная оценка) $\approx$ 31.625

### Задание 2 <a class='anchor' id='statistics_2'>
    
В первом ящике находится 8 мячей, из которых 5 - белые. Во втором ящике - 12 мячей, из которых 5 белых. Из первого ящика вытаскивают случайным образом два мяча, из второго - 4.
    
*Какова вероятность того, что 3 мяча белые?*

Возможны три события:

$C_1$: два белых мяча извлекли из корзины 1, 1 белый - из корзины 2;

$C_2$: 1 белый мяч извлекли из корзины 1, 2 белых - из корзины 2;

$C_3$: все три белых мяча извлекли из корзины 2.

Это несовместные события, поэтому искомая вероятность равна сумме вероятностей отдельных событий:

$$P = P(C_1) + P(C_2) + P(C_3).$$

Рассмотрим каждый вариант отдельно.

**1) $C_1$: 2 белых мяча извлекли из корзины 1, 1 белый - из корзины 2.**

$A$: извлекаем два белых мяча из корзины 1.

Вероятность события $A$ вычислим, используя формулы комбинаторики.

Общее количество исходов:

$$n_A = C_8^2.$$

Количество благоприятных сочетаний для события $A$ (извлекаем два мяча из первой корзины и оба белые): 

$$m_A = С_5^2.$$

Тогда вероятность события $A$ по классическому определению вероятности:

$$P(A) = \frac{m_A}{n_A} = \frac{С_5^2}{C_8^2}.$$

$B$: извлекаем один белый и три черных мяча из корзины 2.

Общее количество исходов:

$$n_B = C_{12}^4.$$

Количество сочетаний для трех черных мячей: $C_7^3$,

оставшийся 1 белый мяч можно извлечь из второй корзины 5 способами: $C_5^1$.

Получаем количество благоприятных сочетаний для события $B$: 

$$m_B = С_5^1 \cdot C_7^3.$$

Вероятность события $B$ по классическому определению вероятности:

$$P(B) = \frac{m_B}{n_B} = \frac{С_5^1 \cdot C_7^3}{C_{12}^4}.$$

Значит вероятность для первого случая:

$$P(C_1) = P(A \cdot B) = P(A) \cdot P(B) = \frac{С_5^2}{C_8^2} \cdot \frac{С_5^1 \cdot C_7^3}{C_{12}^4}.$$

**2) $C_2$: 1 белый мяч извлекли из корзины 1, 2 белых - из корзины 2.**

$A$: извлекаем 1 белый и 1 черный мяч из корзины 1.

Общее количество исходов:

$$n_A = C_8^2.$$

Один белый мяч из корзины 1 можно извлечь 5 способами: $C_5^1$,

один черный мяч можно извлечь 3 способами: $C_3^1$.

Получаем количество благоприятных сочетаний для события $A$: 

$$m_A = С_5^1 \cdot C_3^1.$$

Тогда вероятность события $A$ по классическому определению вероятности:

$$P(A) = \frac{m_A}{n_A} = \frac{С_5^1 \cdot C_3^1}{C_8^2}.$$

$B$: извлекаем 2 белых и 2 черных мяча из корзины 2.

Общее количество исходов:

$$n_B = C_{12}^4.$$

Количество сочетаний для двух белых мячей: $C_5^2$,

количество сочетаний для оставшихся двух черных мячей: $C_7^2$.

Получаем количество благоприятных сочетаний для события $B$: 

$$m_B = С_5^2 \cdot C_7^2.$$

Вероятность события $B$ по классическому определению вероятности:

$$P(B) = \frac{m_B}{n_B} = \frac{С_5^2 \cdot C_7^2}{C_{12}^4}.$$

Значит вероятность для второго случая:

$$P(C_2) = P(A \cdot B) = P(A) \cdot P(B) = \frac{С_5^1 \cdot C_3^1}{C_8^2} \cdot \frac{С_5^2 \cdot C_7^2}{C_{12}^4}.$$

**3) $C_3$: все 3 белых мяча извлекли из корзины 2.**

$A$: извлекаем два черных мяча из корзины 1.

Общее количество исходов:

$$n_A = C_8^2.$$

Количество благоприятных сочетаний для события $A$ (извлекаем два черных мяча из первой корзины): 

$$m_A = С_3^2.$$

Тогда вероятность события $A$ по классическому определению вероятности:

$$P(A) = \frac{m_A}{n_A} = \frac{С_3^2}{C_8^2}.$$

$B$: извлекаем три белых и один черный мяч из корзины 2.

Общее количество исходов:

$$n_B = C_{12}^4.$$

Количество сочетаний для трех белых мячей: $C_5^3$,

оставшийся 1 черный мяч можно извлечь из второй корзины 7 способами: $C_7^1$.

Получаем количество благоприятных сочетаний для события $B$: 

$$m_B = С_5^3 \cdot C_7^1.$$

Вероятность события $B$ по классическому определению вероятности:

$$P(B) = \frac{m_B}{n_B} = \frac{С_5^3 \cdot C_7^1}{C_{12}^4}.$$

Значит вероятность для третьего случая:

$$P(C_3) = P(A \cdot B) = P(A) \cdot P(B) = \frac{С_3^2}{C_8^2} \cdot \frac{С_5^3 \cdot C_7^1}{C_{12}^4}.$$

**Таким образом**, вероятность того, что среди выбранных мячей окажется ровно 3 белых:

$$P = P(C_1) + P(C_2) + P(C_3) \approx 0.369.$$

In [15]:
# 2 белых мяча извлекли из корзины 1, 1 белый - из корзины 2
P_C1_A = combinations(5, 2) / combinations(8, 2)
P_C1_B = combinations(5, 1) * combinations(7, 3) / combinations(12, 4)

P_C1 = P_C1_A * P_C1_B
print(f'P_C1 = {P_C1}')

# 1 белый мяч извлекли из корзины 1, 2 белых - из корзины 2
P_C2_A = combinations(5, 1) * combinations(3, 1) / combinations(8, 2)
P_C2_B = combinations(5, 2) * combinations(7, 2) / combinations(12, 4)

P_C2 = P_C2_A * P_C2_B
print(f'P_C2 = {P_C2}')

# 3 белых мяча извлекли из корзины 2
P_C3_A = combinations(3, 2) / combinations(8, 2)
P_C3_B = combinations(5, 3) * combinations(7, 1) / combinations(12, 4)

P_C3 = P_C3_A * P_C3_B
print(f'P_C3 = {P_C3}')

# Результирующая вероятность
P = P_C1 + P_C2 + P_C3
print(f'P = {P}')

P_C1 = 0.12626262626262627
P_C2 = 0.22727272727272727
P_C3 = 0.01515151515151515
P = 0.3686868686868687


**Ответ (задание 2):** $P \approx 0.369$.

### Задание 3 <a class='anchor' id='statistics_3'>
    
На соревновании по биатлону один из трех спортсменов стреляет и попадает в мишень. Вероятность попадания для первого спортсмена равна 0.9, для второго — 0.8, для третьего — 0.6.

Событие $A$ - стрелок попадает в мишень.

Событие $B_1$ - выстрелил первый спортсмен.

Событие $B_2$ - выстрелил второй спортсмен.

Событие $B_3$ - выстрелил третий спортсмен.

$B_1$, $B_2$, $B_3$ образует полную группу несовместных событий. 

Чтобы определить $P(B_i|A)$ - вероятность события $B_i$, $i=1, 2, 3$ (в мишень попал $i$-ый спортсмен) при условии, что событие $A$ (стрелок попадает в мишень) уже произошло, будет использоваться формула Байеса:

$$P(B_i|A) = \frac{P(B_i) \cdot P(A|B_i)}{P(A)}.$$

Вероятность того, что выстрелил $i-$ый спортсмен, $i=1, 2, 3$:

$$P(B_i) = \frac{1}{3}.$$

Вычислим вероятность попадания в мишень:

$$P(A) = \sum_{i = 1}^3 P(B_i) \cdot P(A|B_i) = \frac{1}{3} (0.9 + 0.8 + 0.6) \approx 0.767. $$

In [16]:
# Вероятность попадаения в мишень по формуле полной вероятности
P_A = (0.9 + 0.8 + 0.6) / 3
P_A

0.7666666666666667

In [17]:
# Переопределим функцию расчета вероятности по формуле Байеса для задачи 3
def bayes_task3(P_A_B):
    return bayes(P_A, P_A_B, 1/3)

**Задание 3.а** <a class='anchor' id='statistics_3_1'>
    
*Найти вероятность того, что выстрел произведен первым спортсменом.*

Используя формулу Байеса, найдем вероятность того, что в мишень попал первый спортсмен:

$$P(B_1|A) = \frac{P(B_1) \cdot P(A|B_1)}{P(A)} = \frac{0.9}{0.9 + 0.8 + 0.6} \approx 0.391.$$

In [18]:
P_B1_A = bayes_task3(0.9)
P_B1_A

0.3913043478260869

**Ответ (задание 3.а):** $P \approx 0.391$.

**Задание 3.б** <a class='anchor' id='statistics_3_2'>
    
*Найти вероятность того, что выстрел произведен вторым спортсменом.*

Используя формулу Байеса, найдем вероятность того, что в мишень попал второй спортсмен:

$$P(B_2|A) = \frac{P(B_2) \cdot P(A|B_2)}{P(A)} = \frac{0.8}{0.9 + 0.8 + 0.6} \approx 0.348.$$

In [19]:
P_B2_A = bayes_task3(0.8)
P_B2_A

0.34782608695652173

**Ответ (задание 3.б):** $P \approx 0.348$.

**Задание 3.в** <a class='anchor' id='statistics_3_3'>
    
*Найти вероятность того, что выстрел произведен третьим спортсменом.*

Используя формулу Байеса, найдем вероятность того, что в мишень попал второй спортсмен:

$$P(B_3|A) = \frac{P(B_3) \cdot P(A|B_3)}{P(A)} = \frac{0.6}{0.9 + 0.8 + 0.6} \approx 0.261.$$

In [20]:
P_B3_A = bayes_task3(0.6)
P_B3_A

0.26086956521739124

**Ответ (задание 3.в):** $P \approx 0.261$.

### Задание 4 <a class='anchor' id='statistics_4'>
    
В университет на факультеты A и B поступило равное количество студентов, а на факультет C студентов поступило столько же, сколько на A и B вместе. Вероятность того, что студент факультета A сдаст первую сессию, равна 0.8. Для студента факультета B эта вероятность равна 0.7, а для студента факультета C - 0.9. Студент сдал первую сессию.

Событие $A$ - студент сдал сессию.

Событие $B_1$ - студент учтится на факультете A.

Событие $B_2$ - студент учтится на факультете B.

Событие $B_3$ - студент учтится на факультете C.

$B_1$, $B_2$, $B_3$ образует полную группу несовместных событий. 

Чтобы определить $P(B_i|A)$ - вероятность события $B_i$, $i=1, 2, 3$ (студент учится на $i$-ом факультете) при условии, что событие $A$ (студент сдал сессию) уже произошло, будет использоваться формула Байеса:

$$P(B_i|A) = \frac{P(B_i) \cdot P(A|B_i)}{P(A)},$$

где $P(B_i)$ - вероятность того, что студент учится на $i$-ом факультете.

Вычислим вероятность обучения студента на каждом из факультетов.

Известно, что на факультете A и B учится равное количество студентов, обозначим это количество через $x$. На факультете C учится столько же, сколько на факультетах A и B вместе, то есть $2x$. Общее количество студентов - $4x$.

Тогда по классическому определению вероятности можно вычислить вероятность того, что студент учится на факультете A или B:

$$P(B_1) = P(B_2) = \frac{x}{4x} = \frac{1}{4},$$

а вероятность того, что студент учится на факультете C:

$$P(B_3) = \frac{2x}{4x} = \frac{1}{2}.$$

Вычислим вероятность того, что студент сдал сессию:

$$P(A) = \sum_{i = 1}^3 P(B_i) \cdot P(A|B_i) = \frac{1}{4} \cdot 0.8 + \frac{1}{4} \cdot 0.7 + \frac{1}{2} \cdot 0.9 = 0.825. $$

In [21]:
# Вероятность сдачи студентом сессии
P_A = 0.8 / 4 + 0.7 / 4 + 0.9 / 2
P_A

0.825

**Задание 4.а** <a class='anchor' id='statistics_4_1'>
    
*Какова вероятность, что он учится на факультете A?*

Используя формулу Байеса, найдем вероятность того, что сдавший сессию студент из факультета A:

$$P(B_1|A) = \frac{P(B_1) \cdot P(A|B_1)}{P(A)} = \frac{0.8}{4 \cdot 0.825} \approx 0.242.$$

In [22]:
P_B1_A = bayes(P_A, 0.8, 1/4)
P_B1_A

0.24242424242424246

**Ответ (задание 4.а):** $P \approx 0.242$.

**Задание 4.б** <a class='anchor' id='statistics_4_2'>
    
*Какова вероятность, что он учится на факультете B?*

Используя формулу Байеса, найдем вероятность того, что сдавший сессию студент из факультета B:

$$P(B_2|A) = \frac{P(B_2) \cdot P(A|B_2)}{P(A)} = \frac{0.7}{4 \cdot 0.825} \approx 0.212.$$

In [23]:
P_B2_A = bayes(P_A, 0.7, 1/4)
P_B2_A

0.21212121212121213

**Ответ (задание 4.б):** $P \approx 0.212$.

**Задание 4.в** <a class='anchor' id='statistics_4_2'>
    
*Какова вероятность, что он учится на факультете C?*

Используя формулу Байеса, найдем вероятность того, что сдавший сессию студент из факультета C:

$$P(B_3|A) = \frac{P(B_3) \cdot P(A|B_3)}{P(A)} = \frac{0.9}{2 \cdot 0.825} \approx 0.545.$$

In [24]:
P_B3_A = bayes(P_A, 0.9, 1/2)
P_B3_A

0.5454545454545455

**Ответ (задание 4.в):** $P \approx 0.545$.

### Задание 5 <a class='anchor' id='statistics_5'>
    
Устройство состоит из трех деталей. Для первой детали вероятность выйти из строя в первый месяц равна 0.1, для второй - 0.2, для третьей - 0.25.

In [25]:
P1 = 0.1
P2 = 0.2
P3 = 0.25

**Задание 5.а** <a class='anchor' id='statistics_5_1'>
    
*Какова вероятность того, что в первый месяц выйдут из строя все детали.*

Событие $A$ - вышла из строя первая деталь.

Событие $B$ - вышла из строя вторая деталь.

Событие $C$ - вышла из строя третья деталь.

Эти события является совместными, они происходят одновременно, значит вероятность того, что вышли из строя все детали вычисляется по формуле:

$$P(A \cdot B \cdot C) = P(A) \cdot P(B) \cdot P(C) = 0.1 \cdot 0.2 \cdot 0.25 = 0.005.$$

In [26]:
# Вероятность того, что все детали выйдут из строя
P_task1 = P1 * P2 * P3
print(f'P_task1 = {P_task1}')

P_task1 = 0.005000000000000001


**Ответ (задание 5.а):** $P = 0.005$.

**Задание 5.б** <a class='anchor' id='statistics_5_2'>
    
*Какова вероятность того, что в первый месяц выйдут из строя только две детали.*

Возможны три события:

$C_1$: только первая деталь не вышла из строя;

$C_2$: только вторая деталь не вышла из строя;

$C_3$: только третья деталь не вышла из строя.

Это несовместные события, поэтому искомая вероятность равна сумме вероятностей отдельных событий:

$$P = P(C_1) + P(C_2) + P(C_3).$$

Рассмотрим каждый вариант отдельно. 

**1) $C_1$: первая деталь не вышла из строя.** 

Событие $A$ - не вышла из строя первая деталь.

Событие $B$ - вышла из строя вторая деталь.

Событие $C$ - вышла из строя третья деталь.

Вероятность для первого случая:

$$P(C_1) = P(A \cdot B \cdot C) = P(A) \cdot P(B) \cdot P(C) = (1 - 0.1) \cdot 0.2 \cdot 0.25 = 0.045.$$

**2) $C_1$: вторая деталь не вышла из строя.** 

Событие $A$ - вышла из строя первая деталь.

Событие $B$ - не вышла из строя вторая деталь.

Событие $C$ - вышла из строя третья деталь.

Вероятность для первого случая:

$$P(C_2) = P(A \cdot B \cdot C) = P(A) \cdot P(B) \cdot P(C) = 0.1 \cdot (1 - 0.2) \cdot 0.25 = 0.02.$$

**3) $C_3$: третья деталь не вышла из строя.** 

Событие $A$ - вышла из строя первая деталь.

Событие $B$ - вышла из строя вторая деталь.

Событие $C$ - не вышла из строя третья деталь.

Вероятность для первого случая:

$$P(C_3) = P(A \cdot B \cdot C) = P(A) \cdot P(B) \cdot P(C) = 0.1 \cdot 0.2 \cdot (1 - 0.25) = 0.015.$$

**Таким образом**, вероятность того, что в первый месяц выйдут из строя только две детали:

$$P = P(C_1) + P(C_2) + P(C_3) = 0.045 + 0.02 + 0.015 = 0.08.$$

In [27]:
# Не вышла из строя только первая деталь
P_C1 = (1 - P1) * P2 * P3
print(f'P_C1 = {P_C1}')

# Не вышла из строя только вторая деталь
P_C2 = P1 * (1 - P2) * P3
print(f'P_C2 = {P_C2}')

# Не вышла из строя только третья деталь
P_C3 = P1 * P2 * (1 - P3)
print(f'P_C3 = {P_C3}')

# Результирующая вероятность того, что выйдет из строя только две детали
P_task2 = P_C1 + P_C2 + P_C3
print(f'P_task2 = {P_task2}')

P_C1 = 0.045000000000000005
P_C2 = 0.020000000000000004
P_C3 = 0.015000000000000003
P_task2 = 0.08


**Ответ (задание 5.б):** $P = 0.08 $.

**Задание 5.в** <a class='anchor' id='statistics_5_3'>
    
*Какова вероятность того, что в первый месяц выйдут из строя хотя бы одна деталь.*

Решим задачу от противного. Найдем вероятность того, что ни одна деталь не вышла из строя в первый месяц, и потом вычтем полученное значение из единицы.

Событие $A$ - не вышла из строя первая деталь.

Событие $B$ - не вышла из строя вторая деталь.

Событие $C$ - не вышла из строя третья деталь.

Эти события является совместными, они происходят одновременно, значит вероятность того, что вышли из строя все детали вычисляется по формуле (берем обратные вероятности от тех, что даны в условии задачи):

$$P(A \cdot B \cdot C) = P(A) \cdot P(B) \cdot P(C) = (1 - 0.1) \cdot (1 - 0.2) \cdot (1 - 0.25) = 0.54.$$

Теперь, зная вероятность обратного события, вычислим вероятность того, что хотя бы одна деталь выйдет из строя:

$$P = 1 - P(A \cdot B \cdot C) = 1 - 0.54 = 0.46.$$

In [28]:
# Вероятность того, что все детали работают исправно, ни одна из них не вышла из строя
P_task3_inv = (1 - P1) * (1 - P2) * (1 - P3)
print(f'P_task3_inv = {P_task3_inv}')

# Результирующая вероятность того, что вышла хотя бы одна деталь из строя
P_task3 = 1 - P_task3_inv
print(f'P_task3 = {P_task3}')

P_task3_inv = 0.54
P_task3 = 0.45999999999999996


**Ответ (задание 5.в):** $P = 0.46 $.

**Задание 5.г** <a class='anchor' id='statistics_5_4'>
    
*Какова вероятность того, что в первый месяц выйдут из строя от одной до двух деталей.*

Возможны два события:

$C_1$: вышла из строя одна деталь;

$C_2$: вышла из строя две детали.

Это несовместные события, поэтому искомая вероятность равна сумме вероятностей отдельных событий:

$$P = P(C_1) + P(C_2).$$

Вероятность события $C_2$ была вычислена в [Задании 5.б](#statistics_5_2):

$$P(C_2) = 0.08.$$

Вычислим вероятность события $C_1$. 

В рамках этого события возможны случаи:

1) вышла из строя первая деталь;

2) вышла из строя вторая деталь;

3) вышла из строя третья деталь.

Сразу запишем вероятность для события $C_1$ (каждое слагаемое соотвествует одному из вариантов события $C_1$, для исправных деталей берется обратная вероятность):

$$P(C_1) = 0.1 \cdot 0.8 \cdot 0.75 + 0.9 \cdot 0.2 \cdot 0.75 + 0.9 \cdot 0.8 \cdot 0.25 = 0.375$$.

Таким образом, результирующая вероятность того, что выйдет из строя одна или две детали:

$$P = P(C_1) + P(C_2) = 0.375 + 0.08 = 0.455.$$

In [29]:
# Вероятность того, что вышла из строя только одна деталь
P_C1 = P1 * (1 - P2) * (1 - P3) + (1 - P1) * P2 * (1 - P3) + (1 - P1) * (1 - P2) * P3
print(f'P_C1 = {P_C1}')

# Результирующая вероятность того, что вышли из строя одна или две детали
P_task4 = P_C1 + P_task2
print(f'P_task4 = {P_task4}')

P_C1 = 0.375
P_task4 = 0.455


**Ответ (задание 5.г):** $P = 0.455$.